src/Juvix/Compiler/Core/Extra.hs

```diff
@@ -116,7 +120,7 @@ developBeta = umap go
 
 etaExpand :: Int -> Node -> Node
 etaExpand 0 n = n
-etaExpand k n = mkLambdas' k (mkApps' (shift k n) (map mkVar' (reverse [0 .. k - 1])))
+etaExpand k n = mkLambdas' k (mkApps' (shift k n) (map mkVar' [0 .. k - 1]))
```

Comment:
The `reverse` was intensional here. You want to eta-expand `f` to `\x \y f x y` not to `\x \y f y x`

Reply:
this change was a consequence of the (incorrect) reverse in mkApps. Will fix it

Reply:
done

Reply:
Not done. Where is the reverse?

Reply:
now it is